src/hotspot/share/gc/g1/heapRegion.inline.hpp

```diff
@@ -167,6 +167,13 @@ inline size_t HeapRegion::block_size(const HeapWord* p, HeapWord* const pb) cons
   return cast_to_oop(p)->size();
 }
 
+inline void HeapRegion::prepare_for_full_gc() {
```

Comment:
I don't believe this API should be part of this class. It appears entirely caller-specific, and it's unclear why the implementation is just a field-update without knowledge of the caller context. While the comment helps mitigate the situation, making it acceptable for a bugfix, I think it would be more appropriate to relocate methods related to parsable-bottom outside of this class.

Reply:
>I don't believe this API should be part of this class. It appears entirely caller-specific,

All of these `_for_full_gc`/`_after_full_gc` methods (and `note_*_marking/scrubbing` methods for concurrent mark) are caller specific, for exactly one caller, for a particular situation the caller encounters. Is there something particularly problematic with this exact method compared to others?

> and it's unclear why the implementation is just a field-update without knowledge of the caller context. While the comment helps mitigate the situation, making it acceptable for a bugfix, I think it would be more appropriate to relocate methods related to parsable-bottom outside of this class.

So basically `_parsable_bottom` (and `_top_at_mark_start` since it is very similar to PB) fields should be somewhere else then? I think there is a CR for moving TAMS somewhere else, there is nothing for PB.

Reply:
> Is there something particularly problematic with this exact method compared to others?

No; all of them require caller-info to understand the impl.

> fields should be somewhere else then?

Yes, because they are constructed/maintained by sth external to heap-region, IMO.